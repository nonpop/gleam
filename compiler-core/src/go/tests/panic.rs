use crate::assert_go;

#[test]
fn bare() {
    assert_go!(
        r#"
fn go() {
  panic
}
"#,
    );
}

#[test]
fn panic_as() {
    assert_go!(
        r#"
fn go() {
  let x = "wibble"
  panic as x
}
"#,
    );
}

#[test]
fn as_expression() {
    assert_go!(
        r#"
fn go(f) {
  let boop = panic
  f(panic)
}
"#,
    );
}

#[test]
fn pipe() {
    assert_go!(
        r#"
fn go(f) {
  f |> panic
}
"#,
    );
}

#[test]
fn sequence() {
    assert_go!(
        r#"
fn go(at_the_disco) {
  panic
  at_the_disco
}
"#,
    );
}

#[test]
fn case() {
    assert_go!(
        r#"
fn go(x) {
  case x {
    _ -> panic
  }
}
"#,
    );
}
