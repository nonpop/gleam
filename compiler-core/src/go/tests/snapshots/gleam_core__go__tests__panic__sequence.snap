---
source: compiler-core/src/go/tests/panic.rs
expression: "\nfn go(at_the_disco) {\n  panic\n  at_the_disco\n}\n"
snapshot_kind: text
---
----- SOURCE CODE

fn go(at_the_disco) {
  panic
  at_the_disco
}


----- COMPILED GO
package mod_P

import (
  gleam_P "example.com/todo/gleam"
)

const _ = gleam_P.Use_Import
const Use_Import byte = 0

func go π[I gleam_P.Type[I]](atTheDisco I) I {
  panic(gleam_P.MakeError(
    "panic",
    "my/mod",
    3,
    "go",
    "`panic` expression evaluated.",
    map[string]any{},
  ))
  return atTheDisco
}
